torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -656,25 +662,81 @@ FusionExecutor::GlobalBuffers FusionExecutor::allocGlobalVals(
 }
 
 std::vector<at::Tensor> FusionExecutor::allocOutputs(
-    const at::ArrayRef<IValue>& inputs,
     kir::ExpressionEvaluator& expr_eval,
     const std::unordered_set<int>& alias_indices) {
   FUSER_PERF_SCOPE("FusionExecutor::AllocOutputs");
   const auto kernel = lowered_->kernel();
   // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
   std::vector<at::Tensor> outputs;
+  for (const auto out_i : c10::irange(kernel->outputs().size())) {
+    // TODO: FIX this short-cut where we trivially forward inputs to outputs
+    if (kernel->outputs()[out_i]->isFusionInput()) {
+      TORCH_INTERNAL_ASSERT(false, "trivial input forwarding NOT IMPLEMENTED");
+      // for (auto inp_i : c10::irange(kernel->inputs().size())) {
+      //   if (kernel->inputs()[inp_i] == kernel->outputs()[out_i]) {
+      //     TORCH_INTERNAL_ASSERT(
+      //         inp_i < inputs.size(),
+      //         "Issue with an input showing up as output, couldn't find
+      //         input.");
+      //     TORCH_INTERNAL_ASSERT(
+      //         inputs[inp_i].isTensor(),
+      //         "Cannot register a scalar as an output in a fusion.");
+      //     outputs.push_back(inputs[inp_i].toTensor());
+      //     break;
+      //   }
+      // }
+    } else {
+      TORCH_INTERNAL_ASSERT(
+          kernel->outputs()[out_i]->isA<TensorView>(),
+          "Cannot allocate outputs that are not tensors.");
+      auto output = kernel->outputs()[out_i]->as<TensorView>();
+      if (alias_indices.count(out_i) != 0) {
+        // aliasing to inputs, no need to allocate real output, just push empty
+        // tensor here.
+        outputs.emplace_back();
+      } else {
+        outputs.push_back(
+            inferAndAllocOutput(output, expr_eval, options_, false));
+      }
+    }
+  }
+  return outputs;
+}
+
+void FusionExecutor::setUsedTVs() {
+  auto used_vals = fusion_->usedMathVals();
+  auto used_tvs = ir_utils::filterByType<TensorView>(used_vals);
+  used_tvs_.insert(used_tvs_.begin(), used_tvs.begin(), used_tvs.end());
```

Comment:
You removed `used_tvs_.clear()` before this call. I think generally we wouldn't be calling this function more than once, but I think it's good to leave in for safety.